starlette/middleware/base.py

```
@@ -156,7 +156,16 @@ async def coro() -> None:
                 if app_exc is not None:
                     nonlocal exception_already_raised
                     exception_already_raised = True
-                    raise app_exc
+                    # Prevent `anyio.EndOfStream` from polluting app exception context.
+                    # If both cause and context are None then the context is suppressed
+                    # and `anyio.EndOfStream` is not present in the exception traceback.
+                    # If exception cause is not None then it is propagated with
+                    # reraising here.
+                    # If exception has no cause but has context set then the context is
+                    # propagated as a cause with the reraise. This is necessary in order
+                    # to prevent `anyio.EndOfStream` from polluting the exception
+                    # context.
+                    raise app_exc from app_exc.__cause__ or app_exc.__context__
```

I guess what happens if the exception has no cause or context? Then does `anyio.EndOfStream` become the new cause / context?

---

If neither the context nor cause are present then this is equivalent to `raise app_exc from None` which suppresses the exception context and just shows the raised exception. This means that `anyio.EndOfStream` does not show up in the traceback.

Should I expand the comment to make this more clear?

This is a desired solution for me since when the `anyio.EndOfStream` was propagated in traceback as the innermost exception (context of app_exc) - it caused our Error Reporting (tool from Google cloud that aggregates encountered errors from logs) to group all our API errors into single group as the innermost exception was always the same.